crates/core/src/host/host_controller.rs

```diff
@@ -233,36 +169,38 @@ impl HostController {
     /// impact the logic of applications. The idea being that if SpacetimeDB is a distributed operating
     /// system, the applications will expect to be called when they are scheduled to be called regardless
     /// of whether the OS has been restarted.
-    pub async fn spawn_module_host(&self, module_host_context: ModuleHostContext) -> Result<ModuleHost, anyhow::Error> {
-        let key = module_host_context.dbic.database_instance_id;
-
-        let (module_host, start_scheduler) = self.make_module_host(module_host_context)?;
+    pub async fn spawn_module_host(&self, mhc: ModuleHostContext) -> Result<ModuleHost, anyhow::Error> {
+        let key = mhc.dbic.database_instance_id;
+
+        let mcc = ModuleCreationContext {
+            dbic: mhc.dbic,
+            scheduler: mhc.scheduler,
+            program_hash: hash_bytes(&mhc.program_bytes),
+            program_bytes: mhc.program_bytes,
+            energy_monitor: self.energy_monitor.clone(),
+        };
+        let module_host = spawn_rayon(move || Self::make_module_host(mhc.host_type, mcc)).await?;
```

Comment:
`make_module_host` doesn't acquire locks?

Reply:
Nope, not afaict